### PR TITLE
WIP First cut of the AIP 2.0 list of RFC versions to be included

### PR DESCRIPTION
**Pull request by swcurran**
Per discussions in recent Aries WG calls, this PR lists the RFCs to be included in the next AIP, pinned at the latest commit in the repo.  Flagged in the PR are the new RFCs added, and those that are still being discussed.  Included are the goals that we used for including RFCs.  Still to do:

- finalize the "Under Consideration"
- review the RFCs that have been included and are not "Accepted" to move them forward.
- update the "Test Suite" section -- what to put in there.
- Add any narrative that might (or might not) be needed about the AIP.  I kind of feel that the RFCs/Versions speak for themselves and by simply being in the AIP, the rationale is implied.

And of course, there is the merging of the PR, and then the merging of a PR to make the AIP v2.0 "Accepted".

### CODE REVIEW

File: concepts/0302-aries-interop-profile/README.md [hunk 1]
```diff
@@ -91,10 +91,69 @@ Feature | [0036-issue-credential](https://github.com/hyperledger/aries-rfcs/tree
...
 
 > To Do: Link(s) to version(s) of the test suite/test cases applicable to this Aries Interop Profile version.
 
+### Aries Interop Profile Version: 2.0
+
+The following are the goals used in selecting RFC versions for inclusion in AIP 2.0, adn the RFCs added as a result of each goal:
+
+- From AIP 1.0: Aries Agents must be able to establish connections, exchange credentials and complete a connection-less proof-request/proof transaction.
+- Aries agents must be able to reuse connections.
+  - RFCs 0434, 0023, 0519, 0360
```
Comment (llorllale) [inline concepts/0302-aries-interop-profile/README.md:104]: > 0360

I submitted a [proposal to move away from did:key](https://github.com/hyperledger/aries-rfcs/pull/577). A primary motivator there is to achieve better cryptographic agility with the ultimate goal of supporting NIST curves.

Supporting NIST curves is very important to us.

File: concepts/0302-aries-interop-profile/README.md [hunk 2]
```diff
@@ -91,10 +91,69 @@ Feature | [0036-issue-credential](https://github.com/hyperledger/aries-rfcs/tree
...
+- Enable access to actionable information in Mobile Agents to enable improvements in the user experience (vs. AIP 1.0-based mobile agents).
+  - RFCs 0183, 0095, 0557
+- Improve support for credential revocation use cases, independent of the revocation mechanism being used.
+  - RFCs 0183, 0441 (Indy AnonCreds specific)
```
Comment (llorllale) [inline concepts/0302-aries-interop-profile/README.md:108]: > 0441

No such RFC?

Comment (swcurran) [reply concepts/0302-aries-interop-profile/README.md:108]: It's here: https://github.com/hyperledger/aries-rfcs/tree/master/concepts/0441-present-proof-best-practices

File: concepts/0302-aries-interop-profile/README.md (hunk 1 above)
Comment (swcurran) [reply concepts/0302-aries-interop-profile/README.md:104]: Are you going to present on this on Wednesday?  This was up in the air when we adjourned before the holidays and you were investigating.

Comment (llorllale) [reply concepts/0302-aries-interop-profile/README.md:104]: @swcurran I can lead a chat in wednesday's call. @dhh1128 is also interested.

**Comment by llorllale** [comment]
We need a common way to ask for credentials and proofs. We should add RFCs 0510 and 0511 for this purpose (both propose using DIF Credential Manifest and Presentation-Exchange).

**Comment by llorllale** [comment]
We also need a way to separate keys based on their purpose (key agreement vs authentication), and we also need to support standard NIST curves/algorithms together with their standardized JOSE envelopes. I propose we add [RFC 0577](https://github.com/hyperledger/aries-rfcs/pull/577) for this.

**Comment by dhh1128** [comment]
>We should add RFCs 0510 and 0511 for this purpose (both propose using DIF Credential Manifest and Presentation-Exchange).

I like the Presentation Exchange work quite a lot. However, I don't think it's stable enough to base this AIP on. I think we will have to do that in the next one.

I do not feel the same way about the Credential Manifest. The problem for me is that it's using a data format to do the job of a protocol. I want to discuss more.

File: concepts/0302-aries-interop-profile/README.md [hunk 3]
```diff
@@ -91,10 +91,69 @@ Feature | [0036-issue-credential](https://github.com/hyperledger/aries-rfcs/tree
...
+Feature | [0334-jwe-envelope](https://github.com/hyperledger/aries-rfcs/tree/be4ad0a6fb2823bb1fc109364c96f077d5d8dffa/features/0334-jwe-envelope) | :question: **Under Consideration**
+Feature | [0434-outofband](https://github.com/hyperledger/aries-rfcs/tree/be4ad0a6fb2823bb1fc109364c96f077d5d8dffa/features/0434-outofband) | :new:
+Feature | [0453-issue-credential-v2](https://github.com/hyperledger/aries-rfcs/tree/be4ad0a6fb2823bb1fc109364c96f077d5d8dffa/features/0453-issue-credential-v2) | AIP V1.0, **Update to V2**
+&nbsp; | Format: Indy AnonCreds | AIP V1.0, Updated
+&nbsp; | Format: LD Signatures | :question: **Under Consideration**
+&nbsp; | Format: BBS+ Signatures | :question: **Under Consideration**
+Feature | [0454-present-proof-v2](https://github.com/hyperledger/aries-rfcs/tree/be4ad0a6fb2823bb1fc109364c96f077d5d8dffa/features/0454-present-proof-v2) | AIP V1.0, **Update to V2**
+&nbsp; | Format: Indy AnonCreds | AIP V1.0, Updated
+&nbsp; | Format: DIF Presentation Exchange | :question: **Under Consideration**
```
Comment (llorllale) [inline concepts/0302-aries-interop-profile/README.md:150]: RFC0510

**Comment by llorllale** [comment]
@dhh1128 

> > We should add RFCs 0510 and 0511 for this purpose (both propose using DIF Credential Manifest and Presentation-Exchange).
> 
> I like the Presentation Exchange work quite a lot. However, I don't think it's stable enough to base this AIP on. I think we will have to do that in the next one.
> 
> I do not feel the same way about the Credential Manifest. The problem for me is that it's using a data format to do the job of a protocol. I want to discuss more.

- After yesterday's call we are all (including you) converging on presentation exchange as a proof request format - good
- I think it is Credential Manifest that may not be stable enough yet. Spec work resumed today and the concepts and use cases are being drawn up again

Do you have a proposal for an interoperable credential request format?

**Comment by dhh1128** [comment]
>Do you have a proposal for an interoperable credential request format?

Yes. My proposal would be to rip out of credential manifest everything that isn't about describing the data of the credential to be issued. No asking the potential holder for data; no description of the issuer; no description of how the credential gets rendered in a UI. All of those things are poison pills to me, but if we could simplify the credential manifest to just what you described -- make it truly a "credential request format", then I would support it.

**Comment by llorllale** [comment]
@swcurran @dhh1128 

`did:peer` is not on this list. What do we need to make that happen?

Can we agree on [method 1](https://identity.foundation/peer-did-method-spec/#generation-method)?

**Comment by dhh1128** [comment]
>Can we agree on method 1?

I don't know who's got support for that method, and whether the support is for static mode only, or static + dynamic. Dynamic support for method 1 would require people to implement [Aries RFC 0030](https://github.com/hyperledger/aries-rfcs/blob/master/features/0030-sync-connection/README.md). I think very few impls have tackled that, and I think that protocol is in need of updating.

Bottom line: I'd be happy to see it adopted, but I think it's a matter for the whole community to weigh in on, so I think we should put it to a vote on both email and a call.

Method 0 would be trivial for anyone who already supports did:key (seriously; it's just modifying a did:key impl to recognize the prefilx `did:peer:0`). Method 1 in static mode is slightly harder, but within impl reach for anyone who can spend a day coding. It's only when you get to Method 1 in dynamic mode that it gets more expensive to support.

**Comment by swcurran** [comment]
@andrewwhitehead -- can you please weigh in on this when you get a chance?  Thanks!

**Comment by llorllale** [comment]
@swcurran @dhh1128 @andrewwhitehead on adopting `did:peer`: #611

**Comment by llorllale** [comment]
I am concerned that we do not have [RFC0348 - Transition Message Type to HTTPs](https://github.com/hyperledger/aries-rfcs/blob/master/features/0348-transition-msg-type-to-https/README.md) in the core section of AIP 2.0.

**Comment by swcurran** [comment]
I'll add it in.  It should be fully executed by now.

@llorllale -- we are adding a config to the AATest Harness so that on startup a test agent knows what AIP they are using, so that we can have intelligent defaults for AIP 2.0 and eliminate some of the need for backward compatibility like this.

File: concepts/0302-aries-interop-profile/README.md [hunk 4]
```diff
@@ -130,102 +130,109 @@ Feature | [0056-service-decorator](https://github.com/hyperledger/aries-rfcs/tre
...
 --- | --- | ---
-Feature | [0510-dif-pres-exch-attach](https://github.com/hyperledger/aries-rfcs/tree/9363159d517afa3c17a79b5d8394e933c652de84/features/0510-dif-pres-exch-attach) | :new
+Feature | [0510-dif-pres-exch-attach](https://github.com/hyperledger/aries-rfcs/tree/b982c24b9083dd5dddff6343dbf534cd1cfe36a6/features/0510-dif-pres-exch-attach) | :new:
 
 #### DIDCOMMV2PREP: DIDComm v2 Prep
  RFC Type | RFC/Link to RFC Version | Note
 --- | --- | ---
-Feature | [0334-jwe-envelope](https://github.com/hyperledger/aries-rfcs/tree/9363159d517afa3c17a79b5d8394e933c652de84/features/0334-jwe-envelope) | :question: **Under Consideration**
-Feature | [0587-encryption-envelope-v2](https://github.com/hyperledger/aries-rfcs/tree/9363159d517afa3c17a79b5d8394e933c652de84/features/0587-encryption-envelope-v2) | :new
+Feature | [0587-encryption-envelope-v2](https://github.com/hyperledger/aries-rfcs/tree/b982c24b9083dd5dddff6343dbf534cd1cfe36a6/features/0587-encryption-envelope-v2) | :new:<br>See envelope note below
```
Comment (dhh1128) [inline concepts/0302-aries-interop-profile/README.md:219]: I'm not sure if this belongs under DIDComm v2 Prep or deserves its own section, but the [formalization of did:peer support in RFC 627](https://github.com/hyperledger/aries-rfcs/tree/master/features/0627-static-peer-dids) may now be worthy of mention?

Comment (swcurran) [reply concepts/0302-aries-interop-profile/README.md:219]: Thanks, Daniel -- I'll get that added.

File: concepts/0302-aries-interop-profile/README.md [hunk 5]
```diff
@@ -91,10 +121,120 @@ Feature | [0036-issue-credential](https://github.com/hyperledger/aries-rfcs/tree
...
+
+#### AIP v2.0 Test Suite
+
+The [Aries Agent Test Harness](https://github.com/hyperledger/aries-agent-test-harness) has a set of tests tagged to exercise AIP 1.0 and AIP 2.0, including the extended targets.
+
+#### Implementers Note about DIDComm Envelopes and the `ACCEPT` element
+
+AIP 2.0 contains two RFCs that reference envelopes 0019-encryption-envelope and 0587-encryption-envelope-v2 (links above).
+The important feature that Aries implementers should understand to differentiate which envelope format can or is being used by an agent is the
+`accept` element of the DIDComm service endpoint. If the `accept` element is not present in the agents service endpoint, the
```
Comment (troyronda) [inline concepts/0302-aries-interop-profile/README.md:234]: @swcurran 
> `accept` element of the DIDComm service endpoint

and the out-of-band message

File: concepts/0302-aries-interop-profile/README.md [hunk 6]
```diff
@@ -91,10 +119,115 @@ Feature | [0036-issue-credential](https://github.com/hyperledger/aries-rfcs/tree
...
+Feature | [0183-revocation-notification](https://github.com/hyperledger/aries-rfcs/tree/d50fe1e38c9720344123146c31fbf7ad8e78b345/features/0183-revocation-notification) | :new:
+Feature | [0360-use-did-key](https://github.com/hyperledger/aries-rfcs/tree/d50fe1e38c9720344123146c31fbf7ad8e78b345/features/0360-use-did-key) | :new:
+Feature | [0434-outofband](https://github.com/hyperledger/aries-rfcs/tree/d50fe1e38c9720344123146c31fbf7ad8e78b345/features/0434-outofband) | :new:
+Feature | [0453-issue-credential-v2](https://github.com/hyperledger/aries-rfcs/tree/d50fe1e38c9720344123146c31fbf7ad8e78b345/features/0453-issue-credential-v2) | Update to V2 Protocol
+Feature | [0454-present-proof-v2](https://github.com/hyperledger/aries-rfcs/tree/d50fe1e38c9720344123146c31fbf7ad8e78b345/features/0454-present-proof-v2) | Update to V2 Protocol
+Feature | [0557-discover-features-v2](https://github.com/hyperledger/aries-rfcs/tree/d50fe1e38c9720344123146c31fbf7ad8e78b345/features/0557-discover-features-v2) | :new:
+Feature | [0627-static-peer-dids](https://github.com/hyperledger/aries-rfcs/tree/d50fe1e38c9720344123146c31fbf7ad8e78b345/features/0627-static-peer-dids)  | :new:
+
```
Comment (troyronda) [inline concepts/0302-aries-interop-profile/README.md:180]: What about RFC092: transport return route? @swcurran @TelegramSam

**Review by swcurran — COMMENTED** [review]
Approved by a unanimous vote at the Aries WG B Call, 2021.05.26.

Requests applied in the final update:

- Added RFC 0092 to the Mediator Coordination sub-target
- Updated the links to the RFC versions to the latest commit in the repo
- Added a link to each RFC in both AIP 1 and 2 to show the diffs between the versions

Merge it @TelegramSam !

**Review by TelegramSam — APPROVED** [review]
Updates match requested in WG meeting

**Pull request merged**